### PR TITLE
Add an insta terminal field and put it in the view.

### PR DESCRIPTION
**Pull request by singularo**
Make a link that when clicked, drops you straight into openshift terminal on a pod.

### CODE REVIEW

File: web/modules/custom/shepherd/shp_orchestration/src/Plugin/OrchestrationProvider/OpenShiftOrchestrationProvider.php [hunk 1]
```diff
@@ -610,6 +611,40 @@ public function getEnvironmentUrl(string $project_name, string $short_name, stri
     }
   }
 
+  public function getTerminalUrl(string $project_name, string $short_name, string $environment_id) {
```
Comment (pingers) [inline web/modules/custom/shepherd/shp_orchestration/src/Plugin/OrchestrationProvider/OpenShiftOrchestrationProvider.php]: Missing inheritdoc.

**Comment by singularo** [comment]
More ready now.

File: web/modules/custom/shepherd/shp_custom/src/Service/Environment.php [hunk 2]
```diff
@@ -239,6 +239,24 @@ public function operationsLinksAlter(array &$operations, EntityInterface $entity
       ];
     }
 
+    $orchestrationProvider = \Drupal::service('plugin.manager.orchestration_provider')
```
Comment (pingers) [inline web/modules/custom/shepherd/shp_custom/src/Service/Environment.php]: Missing comment on this section.

File: web/modules/custom/shepherd/shp_orchestration/src/OrchestrationProviderInterface.php [hunk 3]
```diff
@@ -455,4 +474,14 @@ public function executeJob(
     string $commands = ''
   );
 
+  /**
+   * Get a job.
```
Comment (pingers) [inline web/modules/custom/shepherd/shp_orchestration/src/OrchestrationProviderInterface.php]: Get a bike! :bike: :red_circle: as a :fire_engine:

File: web/modules/custom/shepherd/shp_orchestration/src/Plugin/OrchestrationProvider/DummyOrchestrationProvider.php [hunk 4]
```diff
@@ -236,4 +260,11 @@ public function getEnvironmentUrl(string $project_name, string $short_name, stri
     return [];
   }
 
+  /**
+   * @inheritdoc}
```
Comment (pingers) [inline web/modules/custom/shepherd/shp_orchestration/src/Plugin/OrchestrationProvider/DummyOrchestrationProvider.php]: Missing a brace.

File: web/modules/custom/shepherd/shp_custom/src/Service/Environment.php [hunk 5]
```diff
@@ -239,6 +239,24 @@ public function operationsLinksAlter(array &$operations, EntityInterface $entity
...
+      ->getProviderInstance();
+    $site = $entity->field_shp_site->first()->entity;
```
Comment (fenetikm) [inline web/modules/custom/shepherd/shp_custom/src/Service/Environment.php]: FYI $entity->field_shp_site->entity is the same as $entity->field_shp_site->first()->entity;

Comment (fenetikm) [reply web/modules/custom/shepherd/shp_custom/src/Service/Environment.php]: magic methods ftw

File: web/modules/custom/shepherd/shp_custom/src/Service/Environment.php [hunk 6]
```diff
@@ -239,6 +239,24 @@ public function operationsLinksAlter(array &$operations, EntityInterface $entity
...
+    $project = $site->field_shp_project->first()->entity;
+
+    $terminal = $orchestrationProvider->getTerminalUrl(
```
Comment (fenetikm) [inline web/modules/custom/shepherd/shp_custom/src/Service/Environment.php]: as an alternative to `getTerminalUrl` and the test below on it, you could create a method like `addTerminalOperation` - Tell don't ask. Again, just a thought.

File: web/modules/custom/shepherd/shp_custom/src/Service/Environment.php (hunk 5 above)
Comment (singularo) [reply web/modules/custom/shepherd/shp_custom/src/Service/Environment.php]: Changed.

File: web/modules/custom/shepherd/shp_custom/src/Service/Environment.php (hunk 6 above)
Comment (singularo) [reply web/modules/custom/shepherd/shp_custom/src/Service/Environment.php]: Hmmm later. ;-P

**Comment by pingers** [comment]
Need to run phpcs and I reckon open link in new tab would be nice... it's kinda weird navigating away from Shepherd as is.

**Comment by pingers** [comment]
Mind you, the phpcs issues with Environment.php aren't really the code you added. Meh.

**Pull request merged**